### PR TITLE
Retry project

### PR DESCRIPTION
**Pull request by IsmaPi**
Started from 0

### CODE REVIEW

**Comment by IsmaPi** [comment]
@RaulPPelaez I made a new version, it's not too bad but i'm still working on it

File: scripts/evaluate.py [hunk 1]
```diff
@@ -83,6 +83,11 @@ def main():
     args = parser.parse_args()
     logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
 
+    # Resolve short config names (e.g. "mixed_all" -> "configs/data/mixed_all.yaml")
+    args.model_config = resolve_config_path(args.model_config, "model")
+    if args.data_config:
+        args.data_config = resolve_config_path(args.data_config, "data")
+
```
Comment (RaulPPelaez) [inline scripts/evaluate.py]: I would try not to be "smart" about these things, at least at this point. Provide the program with a correct path or make it fail

File: scripts/train.py [hunk 2]
```diff
@@ -153,6 +154,11 @@ def main():
                         help="Override leave_out_geometry from training config")
     args = parser.parse_args()
 
+    # Resolve short config names (e.g. "mixed_all" -> "configs/data/mixed_all.yaml")
+    args.data_config = resolve_config_path(args.data_config, "data")
+    args.model_config = resolve_config_path(args.model_config, "model")
+    args.train_config = resolve_config_path(args.train_config, "training")
+
```
Comment (RaulPPelaez) [inline scripts/train.py]: same

File: src/utils/config.py [hunk 3]
```diff
@@ -92,6 +93,29 @@ class TrainingConfig:
...
+    """
+    p = Path(value)
+    if p.exists():
+        return value
+    candidate = Path("configs") / config_type / (p.stem + ".yaml")
+    if candidate.exists():
+        return str(candidate)
+    return value
+
+
```
Comment (RaulPPelaez) [inline src/utils/config.py]: I would remove

File: run.py [hunk 4]
```diff
@@ -98,6 +98,12 @@ def main():
     args.data_config = resolve_config_path(args.data_config, "data")
     args.train_config = resolve_config_path(args.train_config, "training")
 
+    # Resolve short config names (e.g. "mixed_all" -> "configs/data/mixed_all.yaml")
+    sys.path.insert(0, str(Path(__file__).parent))
+    from src.utils.config import resolve_config_path
+    args.data_config = resolve_config_path(args.data_config, "data")
+    args.train_config = resolve_config_path(args.train_config, "training")
+
```
Comment (RaulPPelaez) [inline run.py]: same

**Review by RaulPPelaez — COMMENTED** [review]
I do not get what this PR is supposed to do. I see some changes to model loading, and also a notebook with some numbers (and no test loss), but the final test with displacement vs. distance is telling that the model is not producing the correct results.

**Comment by IsmaPi** [comment]
I accidentally closed the PR because of the training logs being exponentially larger and needed to add it to the .gitignore

**Comment by IsmaPi** [comment]
This branch is a revamp of the whole repo. I changed the whole pipeline to ensure that all of it works, and it's what we talked about, creating a PR at the very beginning and then committing to it. The main changes are: full restructuring of the project layout, for example, models were moved from src/models/ to models/, created a utils/ and evaluation/ module, cleaned the repo a little, and deleted useless code (src/models_from_zero/, src/training/, scripts/visualize.py, old evaluation modules). Then i also rewrote both model architectures (torchmd_et.py, torchmd_gn.py) and I'm experimented with delta learning (analytical self-mobility skip connection + learned correction), zero-initialized output heads so training starts from the known physics baseline, and replaced the integer atom-type embedding with a continuous force projection. Furthermore, I implemented the PyTorch Lightning training wrapper (models/lightning_wrapper.py) with proper logging, validation metrics, learning rate scheduling, and early stopping. I also added automated evaluation (evaluation/pair_sweep.py, evaluation/single_step.py, evaluation/plot_training.py) that generates pair-interaction sweep plots, training curves, and single-step error metrics. The notebooks (visualization.ipynb, examples.ipynb) are leftovers from what we did and tested manually on Friday, used to verify that the data pipeline and model loading were working correctly. They are not part of the evaluation pipeline and can be removed. Regarding the pair sweep showing incorrect results: the models are currently not capturing cross-particle hydrodynamic interactions well, the self-mobility baseline (the dominant term) is learned correctly, but the interaction corrections are still being actively worked on  since the results are promising, and I still need to work a bit more on it
